reo/src/outage_costs.py

```diff
@@ -58,12 +58,47 @@ def calc_avoided_outage_costs(data, present_worth_factor, run_uuid):
     site_outputs = data['outputs']['Scenario']['Site']
     pvs = site_outputs['PV']
     wind = site_outputs['Wind']
-    generator = site_outputs['Generator']
+
     load_profile = site_inputs['LoadProfile']
     batt_roundtrip_efficiency = site_inputs['Storage']['internal_efficiency_pct'] \
                                 * site_inputs['Storage']['inverter_efficiency_pct'] \
                                 * site_inputs['Storage']['rectifier_efficiency_pct']
     critical_load = site_outputs['LoadProfile']['critical_load_series_kw']
+
+    """
+    smishra 200515
+    For enabling existing diesel generator for the financial scenario of resilience
+    analysis, the following block is being added. Since the code-flow is same for 
+    both financial and resilience scenarios in a resilience analysis, this block makes
+    sure that existing_kw gets captured for financial case. For resilience scenario, 
+    on the other hand, the generator sizing input to the simulate_outages function must 
+    [capture existing_kw + new_capacity_kw ]
+     
+    Note: "generator_only_runs_during_grid_outage" is set to true by default
+    """
+
+
+    generator_in = site_inputs['Generator']
+    generator_out = site_outputs['Generator']
+
+  # Generator is not allowed to run during normal time
+
+    if generator_in["generator_only_runs_during_grid_outage"]:
+
+        if load_profile.get('outage_start_hour') is None and load_profile.get('outage_end_hour') is None:
+            # handles financial scenario where there will only be non-zero input kw
+            diesel_kw_for_case = generator_in.get('existing_kw')
```

Comment:
@sakshi-mishra should this `.get` and the two below it have a default value of zero? (Or does some later logic require that `diesel_kw_for_case` can be `None`?)

Reply:
@NLaws - yes, the default value for existing_kw is set to zero in the nested_inputs.py.